cpp/src/umap/knn_graph/algo.cuh

```diff
@@ -58,27 +55,20 @@ void launcher(const raft::handle_t& handle,
               const ML::UMAPParams* params,
               cudaStream_t stream);
 
-//  Functor to post-process distances as L2Sqrt*
-template <typename value_idx, typename value_t = float>
-struct DistancePostProcessSqrt : NNDescent::DistEpilogue<value_idx, value_t> {
-  DI value_t operator()(value_t value, value_idx row, value_idx col) const { return sqrtf(value); }
-};
-
 auto get_graph_nnd(const raft::handle_t& handle,
                    const ML::manifold_dense_inputs_t<float>& inputs,
                    const ML::UMAPParams* params)
 {
-  auto epilogue = DistancePostProcessSqrt<int64_t, float>{};
   cudaPointerAttributes attr;
   RAFT_CUDA_TRY(cudaPointerGetAttributes(&attr, inputs.X));
   float* ptr = reinterpret_cast<float*>(attr.devicePointer);
   if (ptr != nullptr) {
     auto dataset =
       raft::make_device_matrix_view<const float, int64_t>(inputs.X, inputs.n, inputs.d);
-    return NNDescent::build<float, int64_t>(handle, params->nn_descent_params, dataset, epilogue);
+    return cuvs::neighbors::nn_descent::build(handle, params->nn_descent_params, dataset);
   } else {
     auto dataset = raft::make_host_matrix_view<const float, int64_t>(inputs.X, inputs.n, inputs.d);
-    return NNDescent::build<float, int64_t>(handle, params->nn_descent_params, dataset, epilogue);
+    return cuvs::neighbors::nn_descent::build(handle, params->nn_descent_params, dataset);
```

Comment:
Does the new NN-descent implementation include an epilogue that adheres to the specified metric?

Reply:
NN Descent in cuVS supports the metric (`L2SqrtExpanded`), so we don't need an epilogue to handle it 👍